### ISSUE TITLE
Fails when startup scripts are inaccessible (even if asked not to modify them)

### ISSUE THREAD

**Issue by goerz** [JuliaLang/juliaup#920]
While preparing setup instructions for a tutorial, I was trying to install `juliaup` on a fresh macOS installation (inside an [UTM virtual machine](https://mac.getutm.app), using the default/automatic installation image provided by Apple for a minimal, unconfigured install). The VM is running Snonoma 14.4.1.

In the standard Terminal app, I tried to install `juliaup` with the recommended `curl -fsSL https://install.julialang.org | sh`, keeping all the default options. The installation failed ("Juliainstaller had a problem and crashed"). The report file boils down to

```
Failed to open file /Users/goerz/.bash_profile

Caused by:
    Permission denied (os error 13)
```

This appears to be related to some kind of macOS security feature, as I get `permission denied` if I try to manually run `touch .bash_profile` in the terminal. Similarly, for other configuration files like `~/.bashrc` and `~/.zshrc`, although `touch random_file` works. Strangely, I haven't been able to find anything on Google about this issue. The only workaround I've found is to run

```
sudo touch .bashrc .bash_profile .zshrc
sudo chown goerz:staff .bashrc .bashr_profile .zshrc
```

to create the files. Once they exist, the `juliaup` installation runs through without any issues.

In principle, I would expect the `juliaup` installer to work around any such macOS permission issues (potentially even by prompting for a sudo password).

Strangely, the crash during the `juliaup` installation also occurs if I customize the installation to decline "Do you want to add the Julia binaries to your PATH by manipulating various shell startup scripts?".

Even if `juliaup` cannot work around the permissions issue, that seems like bug! I would expect not to run into any kind of issue at all related to the startup files if I don't ask the installer to modify the startup file.

**Comment by LilithHafner** [comment]
Applications in mac (even Terminal) often need explicit user permission to modify "files and folders". It's hard to install software without modifying files.

See https://support.apple.com/guide/mac-help/control-access-to-files-and-folders-on-mac-mchld5a35146/mac for a mediocre explanation.

**Comment by goerz** [comment]
Yeah… I actually tried to put Terminal.app into the list of "Full Disk Access" apps, which (surprisingly!) didn't help, either. It's also not one of those special folders like `Desktop`. I actually get a popup asking for confirmation when I try to just `cd` into one of those. That's a little annoying, but okay… it still works. This thing where specific "system" file names inside the home folder are somehow protected (but not other files) seems like something different, and it's not something I've ever seen before, or even something I've been able to find any information about online.
